examples/feature_selection/plot_rfe_digits.py

```diff
@@ -3,34 +3,184 @@
-a digit classification task.
+This example demonstrates how :class:`~sklearn.feature_selection.RFE` can be used
+to determine the importance of individual pixels when classifying handwritten digits.
+RFE is a method that recursively removes the least significant features and retrains
+the model, allowing us to rank features by their importance.
 
 .. note::
 
     See also :ref:`sphx_glr_auto_examples_feature_selection_plot_rfe_with_cross_validation.py`
 
 """  # noqa: E501
 
+# %%
+# Dataset
+# -------
+#
+# We will start by loading the handwritten digits dataset. This dataset consists of 8x8
+# pixel images of handwritten digits. Each pixel will be treated as a feature and we
+# aim to determine which pixels are most relevant for the digit classification task.
+
+# %%
 import matplotlib.pyplot as plt
 
 from sklearn.datasets import load_digits
-from sklearn.feature_selection import RFE
-from sklearn.svm import SVC
 
 # Load the digits dataset
 digits = load_digits()
 X = digits.images.reshape((len(digits.images), -1))
 y = digits.target
 
-# Create the RFE object and rank each pixel
+# Display the first digit
+plt.imshow(digits.images[0], cmap="gray")
+plt.title(f"Label: {digits.target[0]}")
+plt.axis("off")
+plt.show()
+
+# %%
+# Splitting the dataset for evaluation
+# ------------------------------------
+#
+# To assess the benefits of feature selection with
+# :class:`~sklearn.feature_selection.RFE`, we need a training set for selecting
+# features and training our model, and a test set for evaluation.
+# We'll allocate 70% of the data for training and 30% for testing.
+
+# %%
+from sklearn.model_selection import train_test_split
+
+X_train, X_test, y_train, y_test = train_test_split(
+    X, y, test_size=0.3, random_state=42
+)
+
+# %%
+# Benchmarking SVM without Feature Selection
+# ------------------------------------------
+#
+# Before applying :class:`~sklearn.feature_selection.RFE`, let's benchmark the
+# performance of a :class:`~sklearn.svm.SVC` using all features. This will give us
+# a baseline accuracy to compare against.
+
+# %%
+from sklearn.metrics import accuracy_score
+from sklearn.svm import SVC
+
 svc = SVC(kernel="linear", C=1)
-rfe = RFE(estimator=svc, n_features_to_select=1, step=1)
-rfe.fit(X, y)
-ranking = rfe.ranking_.reshape(digits.images[0].shape)
+svc.fit(X_train, y_train)
+y_pred = svc.predict(X_test)
+accuracy_all_features = accuracy_score(y_test, y_pred)
+
+print(f"Accuracy using all {X_train.shape[1]} features: {accuracy_all_features:.4f}")
+
+# %%
+# Feature Selection with RFE
+# --------------------------
+#
+# Now, we'll employ :class:`~sklearn.feature_selection.RFE` to select a subset of
+# the most discriminative features. The goal is to determine if a reduced set of
+# important features can either maintain or even improve the classifier's performance.
+
+# %%
+from sklearn.feature_selection import RFE
+
+# Arbitrarily chosen; can be adjusted based on domain knowledge or iterative testing
```

Comment:
it should be set using a `GridSearchCV` in reality, which you could also do here. You can start by a random number, then optimize using `GridSearchCV`, but putting it in a pipeline with SVC